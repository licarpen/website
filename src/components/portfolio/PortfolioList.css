.PortfolioList{
  margin: 3vw;
  display: grid;
  gap: 30px;
  grid-template-columns: repeat(auto-fit, minmax(330px, 1fr));
  justify-items: center;

  .Link{
    text-decoration: none;
    color: white;
  }

  .title{
    font-size: 20px;
    color: white;
  }

  .card{
    width: 300px;
    height: 335px;
    background: rgb(124, 150, 148);
    padding: 6px;
    padding-top: 15px;
    padding-bottom: 15px;
  }

  .imgDiv{
    height: 270px;
    display: block;
    margin: auto;
    margin-bottom: 10px;
  }

  .card:hover{
    img{
      width: 270px; 
      padding: 0px;
      padding-top: 8px;
    }
    .imgDiv{
      margin-bottom: 10px;
    }
  }
  
  img{
    display: block;
    padding-bottom: 3px;
    margin: auto;
    width: 260px;
    padding: 10px;
    transition: width 0.4s;
  }

  .tech{
    font-size: 18px;
    font-weight: none;
    color:white;
  }

  @media screen and (min-width: 1320px){
    width: 85%;
    margin: auto;
    row-gap: 50px;
  }
}