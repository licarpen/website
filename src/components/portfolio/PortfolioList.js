import React from 'react';
import { Link } from 'react-router-dom';
import { portfolioPieces } from '../../hooks/portfolio';

const PortfolioList = () => {
  const portfolioItemElements = portfolioPieces.map(item => (
    <li key={item.title}>
      <Link to={`/${item.title}`}>
        <h2>{item.title}</h2>
        <img src={item.image}/>
      </Link>
    </li>
    
  ));

  return (
    <ul>
      {portfolioItemElements}
    </ul>
  );
};

export default PortfolioList;

