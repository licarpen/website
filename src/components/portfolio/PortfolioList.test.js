import React from 'react';
import { shallow } from 'enzyme';
import PortfolioList from './PortfolioList';

describe('PortfolioList component', () => {
  it('matches a snapshot when no portfolio piece has been clicked', () => {
    const portfolioItemsArray = [
      { title: 'Treks N Tents',
        image: 'url',
        technology: 'NodeJS, JS, Restful APIs, HTML5/CSS3',
        description: 'A full stack, mobile-responsive hiking and camping app built by a 6-person team in a 4-day sprint' },
      { title: 'Minesweeper No. 2',
        image: 'url',
        technology: 'Vanilla Javascript',
        description: 'A recursion of the classic 90\'s computer game. Built by a 5- person team in a 4-day sprint' },
    ];
    const wrapper = shallow(<PortfolioList portfolioItemDetail='' portfolioItemsArray={portfolioItemsArray}onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('matches a snapshot when a single portfolio item has been clicked and is displayed', () => {
    const portfolioItemsArray = [
      { title: 'Treks N Tents',
        image: 'url',
        technology: 'NodeJS, JS, Restful APIs, HTML5/CSS3',
        description: 'A full stack, mobile-responsive hiking and camping app built by a 6-person team in a 4-day sprint' },
      { title: 'Minesweeper No. 2',
        image: 'url',
        technology: 'Vanilla Javascript',
        description: 'A recursion of the classic 90\'s computer game. Built by a 5- person team in a 4-day sprint' },
    ];
    const wrapper = shallow(<PortfolioList portfolioItemDetail='Treks N Tents' portfolioItemsArray={portfolioItemsArray}onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
