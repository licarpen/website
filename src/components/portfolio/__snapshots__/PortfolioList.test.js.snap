// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioList component matches a snapshot when a single portfolio item has been clicked and is displayed 1`] = `
<ul
  className="PortfolioList"
>
  <li
    className="card"
    key="Plot a Pi"
  >
    <Link
      className="Link"
      to="/Plot a Pi"
    >
      <h2
        className="title"
      >
        Plot a Pi
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        D3, React/Redux, Socket.IO, NodeJS, JS, MongoDB
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Treks N Trails"
  >
    <Link
      className="Link"
      to="/Treks N Trails"
    >
      <h2
        className="title"
      >
        Treks N Trails
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        NodeJS, JS, Restful APIs, PostgreSQL
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Plant Parenthood"
  >
    <Link
      className="Link"
      to="/Plant Parenthood"
    >
      <h2
        className="title"
      >
        Plant Parenthood
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Python, NodeJS, MongoDB, JS, Raspbian
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Postman"
  >
    <Link
      className="Link"
      to="/Postman"
    >
      <h2
        className="title"
      >
        Postman
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        ReactJS
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Computational Models"
  >
    <Link
      className="Link"
      to="/Computational Models"
    >
      <h2
        className="title"
      >
        Computational Models
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Python, ReactJS
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Minesweeper No. 2"
  >
    <Link
      className="Link"
      to="/Minesweeper No. 2"
    >
      <h2
        className="title"
      >
        Minesweeper No. 2
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Vanilla Javascript, HTML5/CSS3
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Data Structures & Algorithms"
  >
    <Link
      className="Link"
      to="/Data Structures & Algorithms"
    >
      <h2
        className="title"
      >
        Data Structures & Algorithms
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        JavaScript
      </p>
    </Link>
  </li>
</ul>
`;

exports[`PortfolioList component matches a snapshot when no portfolio piece has been clicked 1`] = `
<ul
  className="PortfolioList"
>
  <li
    className="card"
    key="Plot a Pi"
  >
    <Link
      className="Link"
      to="/Plot a Pi"
    >
      <h2
        className="title"
      >
        Plot a Pi
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        D3, React/Redux, Socket.IO, NodeJS, JS, MongoDB
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Treks N Trails"
  >
    <Link
      className="Link"
      to="/Treks N Trails"
    >
      <h2
        className="title"
      >
        Treks N Trails
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        NodeJS, JS, Restful APIs, PostgreSQL
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Plant Parenthood"
  >
    <Link
      className="Link"
      to="/Plant Parenthood"
    >
      <h2
        className="title"
      >
        Plant Parenthood
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Python, NodeJS, MongoDB, JS, Raspbian
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Postman"
  >
    <Link
      className="Link"
      to="/Postman"
    >
      <h2
        className="title"
      >
        Postman
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        ReactJS
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Computational Models"
  >
    <Link
      className="Link"
      to="/Computational Models"
    >
      <h2
        className="title"
      >
        Computational Models
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Python, ReactJS
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Minesweeper No. 2"
  >
    <Link
      className="Link"
      to="/Minesweeper No. 2"
    >
      <h2
        className="title"
      >
        Minesweeper No. 2
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        Vanilla Javascript, HTML5/CSS3
      </p>
    </Link>
  </li>
  <li
    className="card"
    key="Data Structures & Algorithms"
  >
    <Link
      className="Link"
      to="/Data Structures & Algorithms"
    >
      <h2
        className="title"
      >
        Data Structures & Algorithms
      </h2>
      <div
        className="imgDiv"
      >
        <img
          src="/path/image"
        />
      </div>
      <p
        className="tech"
      >
        JavaScript
      </p>
    </Link>
  </li>
</ul>
`;
